### DISCUSSION TITLE
Developer documentation

### DISCUSSION THREAD

**Discussion by kirawi** [kovidgoyal/kitty#5961] [General]
Hello! I'm interested in writing my own terminal emulator for fun, and I wanted to reference Kitty's code. I might have missed it, but is there any developer-facing documentation? Otherwise, is there an entry point I can start backtracking from? Thanks.

**Comment by kovidgoyal** [comment]
There's no developer documentation. As for entry points, see parser.c                                                                              
and screen.c                                                                                                                                       
                                                                                                                                                   
However you are probably better of starting with a simpler                                                                                         
terminal to learn from. kitty is probably the most feature rich terminal                                                                           
out there, so there will be more code than in most or all other                                                                                    
terminals.                                                                                                                                         
                                                                                                                                                   
IIRC there is for instance a pure python implementation of a tty I cant                                                                            
remember its name but a bit of googling should find it. That might be                                                                              
easier to get started with.

**Reply by kirawi** [reply]
Awesome, thanks for the response. I'll look around for that.
